src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String SEPARATE_SYMBOL = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String [] data = readFromFile(fromFileName);
+        writeToFile(getReport(data), toFileName);
+    }
+
+    private String [] readFromFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+            return stringBuilder.toString().split(System.lineSeparator());
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can`t open file to read.", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t close file to read.", e);
+        }
```

Comment:
IOException catch is enough, also, add info to the error message to indicate which file we couldn't read